docs/tutorial/index.md

```diff
@@ -0,0 +1,171 @@
+
+## Deploy Charmed Apache Kyuubi K8s
+
+Run the following commands to create a new model, deploy the mandatory charms and relate them:
+
+```shell
+juju add-model tutorial
+
+juju deploy s3-integrator --channel 1/stable
+juju config s3-integrator bucket=mybucket
+juju run s3-integrator/0 sync-s3-credentials access-key=ACCESS_KEY secret-key=SECRET_KEY
+
+juju deploy spark-integration-hub-k8s --channel 3/edge --trust
+juju integrate spark-integration-hub-k8s s3-integrator
+
+juju deploy postgresql-k8s --channel 14/stable --trust
+
+juju deploy data-integrator --channel latest/edge --config database-name=test
+
+juju deploy kyuubi-k8s --channel 3.4/edge --trust
+juju integrate kyuubi-k8s spark-integration-hub-k8s 
+juju integrate kyuubi-k8s:auth-db postgresql-k8s
+juju integrate kyuubi-k8s data-integrator
+```
+
+> **Note**: You may use a different object storage, such as `azure-storage-integrator`.
+
+## Access Charmed Apache Kyuubi K8s
+
+Get the JDBC endpoint and its credentials with the following command:
+
+```shell
+juju run data-integrator/0 get-credentials
+```
+
+```yaml
+kyuubi:
+  data: '{"database": "test", "external-node-connectivity": "true", "provided-secrets":
+    "[\"mtls-cert\"]", "requested-secrets": "[\"username\", \"password\", \"tls\",
+    \"tls-ca\", \"uris\", \"read-only-uris\"]"}'
+  database: test
+  endpoints: 10.64.140.43:10009
+  password: 31rwWzk8wpnhoZvU
+  tls: "False"
+  uris: jdbc:hive2://10.64.140.43:10009/
+  username: relation_id_15
+  version: 1.10.2
+ok: "True"
+```
+
+You may use the endpoint with a JDBC-compliant client, such as `beeline`.
+
+```{note}
+We recommend using the [spark-client](https://snapcraft.io/spark-client) snap, which exposes a beeline client under the `spark-client.beeline` command.
+```
+
+```shell
+spark-client.beeline -u "jdbc:hive2://10.64.140.43:10009/" -n relation_id_15 -p 31rwWzk8wpnhoZvU
+```
+
+## Enable encryption with TLS
+
+[Transport Layer Security (TLS)](https://en.wikipedia.org/wiki/Transport_Layer_Security) is a protocol used to encrypt data exchanged between two applications.
+Essentially, it secures data transmitted over a network.
+
+Typically, enabling TLS between a highly available database and client/server applications requires a high level of expertise.
+This has all been encoded into Charmed Apache Kyuubi K8s so that configuring TLS requires minimal effort on your end.
+
+TLS is enabled by integrating Charmed Apache Kyuubi K8s with the [Self-signed certificates charm](https://charmhub.io/self-signed-certificates).
+This charm centralises TLS certificate management consistently and handles operations like providing, requesting, and renewing TLS certificates.
+
+```{caution}
+**[Self-signed certificates](https://en.wikipedia.org/wiki/Self-signed_certificate) are not recommended for a production environment.**
+
+Check [this guide](https://discourse.charmhub.io/t/security-with-x-509-certificates/11664) for an overview of the TLS certificates charms available.
```

Comment:
Suggestion: Wording improvements.

```suggestion
We recommend to avoid using self-signed TLS certificates for production environments.
Please refer to the [X.509 certificates post](https://charmhub.io/topics/security-with-x-509-certificates) for an overview of the TLS certificates Providers charms and some guidance on how to choose the right charm for your use case.
```